lib/assets/javascripts/react_ujs_mount.js

```diff
@@ -53,7 +53,7 @@
 
         // Assume className is simple and can be found at top-level (window).
         // Fallback to eval to handle cases like 'My.React.ComponentName'.
-        var constructor = window[className] || eval.call(window, className);
+        var constructor = window[className].default || window[className] || eval.call(window, className);
```

Comment:
Yup, it sure would. Sorry about the delay on this. I think the solution you outlined above would work just fine. I'll update and resubmit.